### DISCUSSION TITLE
How to depend on packages from other buckets?

### DISCUSSION THREAD

**Discussion by bharatvaj** [ScoopInstaller/Scoop#6105] [Help and Questions]
Following this issue: https://github.com/ScoopInstaller/Main/issues/5897

It seems that the version of openssl (openssl-light) required for lynx, is in the bucket `versions`.

Locally, I just did `scoop bucket add versions` and changed the manifest of lynx.json to depend on openssl-light and it works.

Though this won't work for other users if they have only main bucket installed. Is there any way in the manifest file, to select a package from a particular bucket, such as: `@versions/openssl-light`?

**Comment by HUMORCE** [comment]
https://github.com/ScoopInstaller/Main/commit/321306d7720a73df4b91d7ed5977d81c30223bd9

Lynx declares that it requires Win32/32bit OpenSSL 3.0.x lib.

- We must hardcode the Win32 OpenSSL download url on the manifest.
- Even 3.3.x works, but 3.0.x are selected.

**Reply by bharatvaj** [reply]
Thanks, @HUMORCE, lynx is working again.

openssl-light is already updated in a manifest in versions bucket. Can we introduce a mechanism to select versions and architectures within the manifest file?

I am trying to write a manifest for a cross-compiler. In that case, we need binaries(binutils) for the host(x86) and libraries for the target architecture, which may be already be packaged in a different manifest specified for arm. I feel like it may be a maintenance burden without these mechanisms.

**Reply by HUMORCE** [reply]
> I am trying to write a manifest for a cross-compiler

In the case, it is better to create different manifest for different architectures of the same package separately.

> Can we introduce a mechanism to select versions and architectures within the manifest file?

You can make a feature request for it, I remember it seems to already exist in one of the feature requests, please try searching for it first.
